src/deadline/job_attachments/download.py

```diff
@@ -1313,3 +1357,133 @@ def download_job_output(
+    session action ID (e.g., .../step_id/task_id/20241225T120000_sessionaction_id/).
+
+    For task-based paths, searches in the task folder first (...job_id/step_id/task_id/) for efficiency.
+    This approach is more correct than searching by timestamp prefix since latestSessionActionId is
+    typically obtained from GetTask API, guaranteeing the latest outputs.
+    For chunked steps, when no task-based manifests found, falls back to step folder (...job_id/step_id/).
+    """
+    outputs: dict[str, list[BaseAssetManifest]] = defaultdict(list)
+
+    # Try task-specific prefix first for efficiency
+    task_prefix = _get_output_manifest_prefix(
+        s3_settings, farm_id, queue_id, job_id, step_id, task_id
+    )
+
+    try:
+        manifests_keys = _get_manifests_by_session_action_regex(
+            task_prefix, s3_settings.s3BucketName, session_action_id, session
+        )
+    except JobAttachmentsError:
+        return outputs
+
+    # If no manifests found at task level, fall back to step level
+    if not manifests_keys:
+        step_prefix = _get_output_manifest_prefix(s3_settings, farm_id, queue_id, job_id, step_id)
+        try:
+            manifests_keys = _get_manifests_by_session_action_regex(
+                step_prefix, s3_settings.s3BucketName, session_action_id, session
+            )
+        except JobAttachmentsError:
+            return outputs
+
+    # Download all found manifests
+    with concurrent.futures.ThreadPoolExecutor(max_workers=S3_DOWNLOAD_MAX_CONCURRENCY) as executor:
+        futures = [
+            executor.submit(
+                get_asset_root_and_manifest_from_s3, key, s3_settings.s3BucketName, session
+            )
+            for key in manifests_keys
+        ]
+        for i, future in enumerate(futures):
+            asset_root, manifest = future.result()
+            if not asset_root:
+                raise MissingAssetRootError(
+                    f"Failed to get asset root from metadata of output manifest: {manifests_keys[i]}"
+                )
+            outputs[asset_root].append(manifest)
+
+    return outputs
+
+
+def _get_manifests_by_session_action_regex(
+    manifest_prefix: str,
+    s3_bucket: str,
+    session_action_id: str,
+    session: Optional[boto3.Session] = None,
+) -> List[str]:
+    """
+    Get manifest keys matching a specific session action ID using regex search.
+
+    Lists all objects under the given prefix and uses regex to find paths containing
+    the session action ID. This is necessary because the exact timestamp prefix in
+    the path is unknown.
+    """
+    manifests_keys = []
+    s3_client = get_s3_client(session=session)
+
+    try:
+        paginator = s3_client.get_paginator("list_objects_v2")
+        page_iterator = paginator.paginate(
+            Bucket=s3_bucket,
+            Prefix=manifest_prefix,
+        )
+
+        regex_pattern = rf".*{re.escape(session_action_id)}.*output.*"
```

Comment:
You're already doing pre-processing, best to use https://docs.python.org/3/library/re.html#re.compile and avoid re-compiling the regex repeatedly for every `re.search` call.

Reply:
Switched to `re.compile`.